File: models/otp.js
const mongoose = require('mongoose');

const otpSchema = new mongoose.Schema({
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    issuer: String,
    label: String,
    secret: String,
    algorithm: String,
    digits: Number,
    period: Number,
    counter: Number,
    method: String,
    createdAt: { type: Date, default: Date.now }
});

const Otp = mongoose.model('Otp', otpSchema);

module.exports = Otp;

File: server/internal/app/models/otp.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type OTP struct {
	ID        primitive.ObjectID `bson:"_id,omitempty"`
	UserID    primitive.ObjectID `bson:"user_id"`
	IssuerID  primitive.ObjectID `bson:"issuer_id"`
	Label     string             `bson:"label"`
	Secret    string             `bson:"secret"`
	Algorithm string             `bson:"algorithm"`
	Digits    int                `bson:"digits"`
	Period    int                `bson:"period"`
	Counter   int                `bson:"counter"`
	Method    string             `bson:"method"`
	CreatedAt time.Time          `bson:"created_at"`
}
